src/dealer/DealerWallet.ts

```diff
@@ -0,0 +1,515 @@
+    // same address is returned each time
+    const { address } = await this.exchange.fetchDepositAddress("BTC")
+    return address
+  }
+
+  async satsBalance() {
+    const { satsLnd: nodeLiabilities } = await this.getLocalLiabilities()
+    const node = -nodeLiabilities
+
+    // at least on FTX. interest will be charged when below -$30,000.
+    // TODO: manage this part
+
+    const { sats: exchange } = await this.getExchangeBalance()
+    const total = node + exchange
+
+    return {
+      total,
+      node,
+      exchange,
+    }
+  }
+
+  async getProfit() {
+    const { total: sats } = await this.satsBalance()
+    const usdAssetsInBtc = sats * UserWallet.lastPrice
+
+    const { usd: usdLiabilities } = await this.getLocalLiabilities()
+
+    const { usdPnl } = await this.getExchangeBalance()
+
+    const usdProfit = usdAssetsInBtc + usdPnl - usdLiabilities
+
+    return {
+      usdProfit,
+    }
+  }
+
+  async getExchangeBalance() {
+    const balance = await this.exchange.fetchBalance()
+    this.logger.debug({ balance }, "this.exchange.fetchBalance result")
+    return {
+      sats: btc2sat(balance.total.BTC ?? 0),
+      usdPnl: balance.total.USD,
+    }
+  }
+
+  async getNextFundingRate() {
+    const {
+      result: { nextFundingRate },
+    } = await this.exchange.getNextFundingRate()
+    return nextFundingRate
+  }
+
+  async getAccountPosition() {
+    // FIXME this helper function is inverse?
+    // or because price = usd/btc, usd/sats, sat or btc are in the denominator
+    // and therefore the "inverse" make sense...?
+    const btcPrice = btc2sat(UserWallet.lastPrice)
+
+    this.logger.debug({ btcPrice }, "[DEBUG] is price set for the dealer?")
+
+    // TODO: what is being returned if no order had been placed?
+    // probably an empty array
+
+    // tmp for log
+    const result = await this.exchange.privateGetAccount()
+    this.logger.debug({ result }, "full result of this.exchange.privateGetAccount")
+
+    const {
+      result: {
+        collateral,
+        positions,
+        chargeInterestOnNegativeUsd,
+        marginFraction,
+        totalAccountValue,
+      },
+    } = result
+    this.logger.debug(
+      { collateral, positions, chargeInterestOnNegativeUsd, marginFraction },
+      "value kept from this.exchange.privateGetAccount",
+    )
+
+    // getPositions() instead ?!?
+    const positionBtcPerp = _.find(positions, { future: this.exchange.symbol })
+    this.logger.debug({ positionBtcPerp }, "positionBtcPerp result")
+
+    const {
+      netSize = 0,
+      estimatedLiquidationPrice,
+      collateralUsed,
+      maintenanceMarginRequirement,
+    } = positionBtcPerp ?? {}
+
+    // TODO: check this is the intended settings
+    assert(chargeInterestOnNegativeUsd === true)
+
+    assert(netSize <= 0)
+
+    return {
+      // making netSize positive to simplify calculation
+      // we should always be short so netSize should initially
+      // always be negative
+      btc: -netSize,
```

Comment:
Instead of a comment can we codify this expectation via a function name or type?
This kind of implicit conversion can be dangerous and introduce subtle bugs if somebody isn't aware or forgets.
Some ideas how to improve it:
Instead of `btc` call the value: `btcShortPosition` or something else more descriptive.
Instead of `~` create an explicit function `normalizePositionSize`. Even if its just 1 line it'll help communicate what we are doing.
Perhaps introduce a wrapper type `NormalizedPositionSize(int)` that only gets returned from that function.

I think this approach will reduce the likelihood of introducing errors later on.